logging_utilities/filters/type_validator.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+
+def is_instance(attr, of_type):
+    if isinstance(of_type, str):
+        attr_class = type(attr)
+        if '.' in of_type:
+            return attr_class.__module__ + '.' + attr_class.__name__ == of_type or of_type in [
+                c.__module__ + '.' + c.__name__ for c in attr_class.__bases__
+            ]
+        return attr_class.__name__ == of_type or of_type in [
+            c.__name__ for c in attr_class.__bases__
+        ]
+    return isinstance(attr, of_type)
+
+
+class TypeValidator(logging.Filter):
+    """Filter attributes based on their types
+
+    This filter can help in case multiple libraries/frameworks etc. use the
+    same extra properties in the extra parameter of the logging. It filters
+    these extra properties by type, either with a whitelist (the default) or
+    with a blacklist.
+    """
+
+    def __init__(self, typecheck_list, *, is_blacklist=False):
```

Comment:
Why `, *,` ?

Reply:
It's not absolutely necessary. But passing such a boolean flag as positional parameter would make the invocation very hard to read, so I thought why not require "is_blacklist" to be a keyword argument.

Reply:
keyword argument is a good idea, but my question why to add `*,` and not simply `def __init__(self, typecheck_list, is_blacklist=False):` ?